### PR TITLE
[Parser] Keep source location info for the ownership keywords inside a closure capture list

### PR DESCRIPTION
**Pull request by akyrtzi**
This allows to do syntax coloring for 'weak'/'unowned' inside a capture list.
rdar://42655051

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci smoke test and merge

File: test/IDE/coloring.swift [hunk 1]
```diff
@@ -390,3 +390,14 @@ func keywordInCaseAndLocalArgLabel(_ for: Int, for in: Int, class _: Int) {
...
+// CHECK: <kw>class</kw> Ownership {
+class Ownership {
+  // CHECK: <attr-builtin>weak</attr-builtin> <kw>var</kw> w
+  weak var w
+  // CHECK: <attr-builtin>unowned</attr-builtin> <kw>var</kw> u
+  unowned var u
+}
+// CHECK-OLD: <kw>let</kw> closure = { [<attr-builtin>weak</attr-builtin> x=bindtox, <attr-builtin>unowned</attr-builtin> y=bindtoy] <kw>in</kw> }
+// FIXME: CHECK-NEW: <kw>let</kw> closure = { [weak x=bindtox, unowned y=bindtoy] <kw>in</kw> }
+let closure = { [weak x=bindtox, unowned y=bindtoy] in }
```
Comment (rintaro) [inline test/IDE/coloring.swift]: How about `unowned(unsafe)`?

Comment (akyrtzi) [reply test/IDE/coloring.swift]: Thanks for pointing it out! Made change to keep the source range for that.

**Comment by akyrtzi** [comment]
@swift-ci smoke test

**Review by rintaro — APPROVED** [review]
Looks good. Thank you!

**Pull request merged**